server/src/main/java/org/elasticsearch/action/bulk/BackoffPolicy.java

```diff
@@ -43,6 +45,56 @@
  */
 public abstract class BackoffPolicy implements Iterable<TimeValue> {
 
+    private static final int DEFAULT_RETRY_LIMIT = 10;
```

Comment:
I split the two dynamic limits into unlimited and limited. The unlimited is used for potentially long running operations (insert from query, copy, where we do more dynamic throttling as we can pause the reads). The limited variant is used for insert-from-values where we may not want to retry for too long and instead error out to apply back-pressure to clients.